app_history.d.ts

```diff
@@ -2,6 +2,11 @@ interface Window {
   readonly appHistory: AppHistory;
 }
 
+interface AppHistoryEventMap {
+  "navigate": AppHistoryNavigateEvent;
+  "navigateerror": ErrorEvent;
```

Comment:
Should we also add `navigatesuccess`? That's just `Event`, but it looks like that's still reasonably common: e.g. https://github.com/microsoft/TypeScript/blob/9cc30709380da8b212487f63d134b7be9b001511/lib/lib.dom.d.ts#L1916-L1931

Reply:
Done